pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -657,7 +657,19 @@ func (c *Controller) getForeignServiceInstancesByPort(svc *model.Service, reqSvc
 
 	selector := labels.Instance(svc.Attributes.LabelSelectors)
 
-	// Get the service port name so that we can construct the service instance
+	// Get the service port name and target port so that we can construct the service instance
+	item, exists, err := c.serviceInformer.GetStore().GetByKey(kube.KeyFunc(svc.Attributes.Name, svc.Attributes.Namespace))
+	// We did not find the k8s service. We cannot get the targetPort
```

Comment:
meta comment: I investigated this recently and I couldn't find any way this function actually returns an error. Still good to handle it though